Toward vuejs v2.0

lionel-bijaoui opened the pull request:
🚧 This is WIP 🚧 
current problem by order of importance:
- Tests need to be rewritten in some way to fit vuejs 2.0 😅 
- selectEx: weird warning, problem is not yet identified.
- dev app: is not working well and can be simplified with ES6.
- validator: weird warning seem to indicate lack of information on initialisation. May be related to dev app or internal problem with vuejs v2.0. Some behavior have changed and things update faster which may or may not be the cause.
- noUiSlider: I'm updating it to v9.0 and there is breaking changes and cool new functionality (very useful for my work).
- vue multiselect should work now that #87 is implemented, but I need to make sure.
- updating plugins

Project board : https://github.com/icebob/vue-form-generator/projects/2

icebob commented on .babelrc:
```diff
@@ -1,5 +1,5 @@
 {
-  "presets": ["es2015", "stage-0"],
+  "presets": ["es2015", "stage-2"],
   "plugins": ["transform-runtime"],
```
Why changed to stage-2 from 0?

lionel-bijaoui replied:
stage-2 is the default in the vue-cli webpack preset. It is less bleeding edge, I guess ? I don't mind putting it back to were it was, I just did that when I was debugging to exclude differences between your config and the proposed webpack config.
You use ES7 features ?

icebob replied:
Currently not, just I'm using stage-0 in my other projects too.

icebob commented on src/fields/abstractField.js:
```diff
@@ -1,93 +1,100 @@
+
+                if (isFunction(this.schema.set)) {
+                    this.schema.set(this.model, newValue);
+                    console.log("model-updated via schema", this.model[this.schema.model]);
```
Not guaranteed, that this.schema.model is defined. Need to check

lionel-bijaoui replied:
BTW, what is this `schema.set` ? I might have missed it, but I don't recall any explanation about it in the doc.

icebob commented on webpack.dev.config.js:
```diff
@@ -1,27 +1,21 @@
-		"test": /\.jade?$/,
-		"loader": "jade"
-	},
-	{
```
In Vue2 these loaders not neccessary?

lionel-bijaoui replied:
Well, we don't require any `.jade`, `.scss` or `.css` files. They were never necessary (vue-loader take care of this).
It is also part of my temporary debug changes.
It can came back we everything is working again.

icebob replied:
Hmm, you are right. It's a habit :)

icebob commented on test/unit/specs/VueFormGenerator.spec.js:
```diff
@@ -10,24 +10,27 @@ let el, vm;
 
 describe("VueFormGenerator.vue", () => {
-
+	describe.only("nothing", () => {
```
It's misleading, because Travis will be green, but no correct testing. In development state there is no problem if Travis red, but show the real results and we know we need to fix it too. :wink:

lionel-bijaoui replied:
Sorry about that, it was to test the build locally without changing too much. Normally, it is forbidden to push `.only()`. This will go away when I fix the test. I forget to put it back how it was.

icebob replied:
OK

icebob commented:
Hi, are you working on this upgrade yet?

lionel-bijaoui commented:
@icebob Sorry I was very busy at work. I will work on it soon.

icebob commented:
@lionel-bijaoui Do you have any new commit to this PR?
Because maybe next week I'll able to work on vue2 migration. And I want to merge this PR to the `next` branch.

lionel-bijaoui commented:
I'm sorry, I'm really squeezed tight at work. I'm not in my normal office and I'm not doing any dev work. It's kinda special for me for the next 2 months.
I'm going to take a look at my PR tomorrow. Worse case scenario, you can use my work as a base. Most of it work well. The unit tests are the biggest job along the validator system.
Sorry I can't be more helpful, I feel really bad about that 😞

icebob commented:
Thanks, no problem! Just upload your last state and I continue to migration. :wink:

lionel-bijaoui commented:
I updated vue and integrated all the new changes since the last push. The main problem is the validator that is validating too early. I don't know if it has access to the values or not.
With that, the test need to be fixed and updated.
Lastly, some libs were updated and need to be checked for changes (NoUiSlider for example can now accept an arbitrary number of handle + syntax changes)

icebob commented:
Thank you!
